SAP/open-ux-tools
# feat(axios-extension): add support for login with reentrance ticket (Cloud ABAP)

This PR adds support to connect to Embedded Steampunk systems using [reentrance tickets](https://launchpad.support.sap.com/#/notes/2822156). 

The authentication flow can be summarized as:
1. The user's default browser is opened so that the user can login to the system using SAML 2.0. 
2. After the user is authenticated, the backend system issues a reentrance ticket that is passed back to the code. Reentrance tickets are short-lived (expiry time of 2 minutes) one-time tokens that the backend systems issue. 
3. These tickets are passed back in `MYSAPSSO2` header to fetch cookies that are used to continue the session. 

The user will need to authenticate using the browser _every_ time with this method.

The example CLI code demonstrates how the API is used:
https://github.com/SAP/open-ux-tools/blob/0db5b7f8555a99473a75a2427b6d66dd4be183ab/examples/odata-cli/src/index.ts#L131-L135

The starting point for the code to handle fetch and use the reentrance ticket is here:https://github.com/SAP/open-ux-tools/blob/0db5b7f8555a99473a75a2427b6d66dd4be183ab/packages/axios-extension/src/auth/index.ts#L61-L69

## To test
There is an example module demonstrating how the API can be used: `examples/odata-cli`. The README.md file explains the pre-requisites. Once that is done, at a terminal, change your working directory to `examples/odata-cli` and:
```shell
pnpm build
pnpm start -- cloud
```
You should see a file `v2-catalog.json` created under the `.tmp` folder (re/move it if already exists first).
